libbeat/processors/registry.go

```diff
@@ -1,10 +1,39 @@
 package processors
 
 import (
+	"errors"
+
 	"github.com/elastic/beats/libbeat/common"
 	"github.com/elastic/beats/libbeat/logp"
+	p "github.com/elastic/beats/libbeat/plugin"
 )
 
+type processorPlugin struct {
+	name   string
+	constr Constructor
+}
+
+var pluginKey = "libbeat-processor"
```

Comment:
It doesn't really matter, as the pluginKey is private to the package providing a plugin type. It's only required for them to be 'unique'. We could even use a generated UUID.